### PR TITLE
Check if_exists options

### PR DESCRIPTION
**Pull request by Jesus89**
- Check if_exists options (fail, replace, append)
- Refactor encode_geometry function
- Add param validation unit tests

### CODE REVIEW

File: cartoframes/core/managers/context_manager.py [hunk 1]
```diff
@@ -57,6 +58,9 @@ def copy_from(self, cdf, table_name, if_exists='fail', cartodbfy=True, log_enabl
                             'Please choose a different `table_name` or use '
                             'if_exists="replace" to overwrite it'.format(
                                 table_name=table_name, schema=schema))
+        else:
+            # 'append'
```
Comment (alrocar) [inline cartoframes/core/managers/context_manager.py]: A nice case where a comment adds meaningful information. Thanks ❤️

File: cartoframes/io/carto.py [hunk 2]
```diff
@@ -227,14 +232,16 @@ def copy_table(table_name, new_table_name, credentials=None, if_exists='fail', l
         new_table_name(str, optional): name for the new table.
         credentials (:py:class:`Credentials <cartoframes.auth.Credentials>`, optional):
             instance of Credentials (username, api_key, etc).
-        if_exists (str, optional): 'fail', 'replace'. Default is 'fail'.
+        if_exists (str, optional): 'fail', 'replace', 'append'. Default is 'fail'.
     """
     if not isinstance(table_name, str):
-        raise ValueError('Wrong table name. You should provide a valid string.')
+        raise ValueError('Wrong table name. You should provide a valid table name.')
```
Comment (alrocar) [inline cartoframes/io/carto.py]: Could we add a `:raises ValueError: When the table name provided is wrong or the `if_exists` param is not valid` to the apidocs?

File: cartoframes/io/carto.py [hunk 3]
```diff
@@ -227,14 +232,16 @@ def copy_table(table_name, new_table_name, credentials=None, if_exists='fail', l
...
 
     if not isinstance(new_table_name, str):
-        raise ValueError('Wrong new table name. You should provide a valid string.')
-    pass
+        raise ValueError('Wrong new table name. You should provide a valid table name.')
+
+    if if_exists not in IF_EXISTS_OPTIONS:
+        raise ValueError('Wrong option. You should provide: {}.'.format(', '.join(IF_EXISTS_OPTIONS)))
```
Comment (alrocar) [inline cartoframes/io/carto.py]:
```suggestion
        raise ValueError('Wrong option for the `if_exists` param. You should provide: {}.'.format(', '.join(IF_EXISTS_OPTIONS)))
```

File: cartoframes/utils/geom_utils.py [hunk 4]
```diff
@@ -127,6 +127,17 @@ def _extract_srid(egeom):
         return (0, egeom)
 
 
+def encode_geometry(geom, srid=4326):
+    """Encode geometry: wkt or wkb"""
+    wkt = True
+    if isinstance(geom, shapely.geometry.base.BaseGeometry):
+        if wkt:
```
Comment (alrocar) [inline cartoframes/utils/geom_utils.py]: mmm, I don't get this condition, `wkt` is always True, right?

File: tests/unit/io/test_carto.py [hunk 5]
```diff
@@ -181,3 +175,123 @@ def test_read_carto_decode_geom_false(mocker):
...
+                           'instance or use the `set_default_credentials` function.'
+
+
+def test_to_carto_wrong_if_exists(mocker):
+    # Given
+    df = GeoDataFrame({'geometry': [Point([0, 0])]})
+
+    # When
+    with pytest.raises(ValueError) as e:
+        to_carto(df, '__table_name__', if_exists='keep_calm')
```
Comment (alrocar) [inline tests/unit/io/test_carto.py]: keep_calm... `and carry on`

File: cartoframes/utils/geom_utils.py (hunk 4 above)
Comment (Jesus89) [reply cartoframes/utils/geom_utils.py]: Sorry, I left the flag. I will split this into two functions.

I have tested both ways of uploading data:
- `EWKT`
- `EWKB hex`

There is no clear winner :S. The second one is ~5% smaller, but it is not a big win. Do you think there is an advantage in the server-side using one over the other?

Comment (alrocar) [reply cartoframes/utils/geom_utils.py]: I'd vote for `EKWB hex` if it's smaller. Also I guess that format is closer to the one PostGIS uses to store geometries (if not the same) so we might be saving computing cycles as well.

**Comment by alrocar** [comment]
acceptance 🍏 

I've resolved conflicts and waiting for green tests to merge to develop

**Pull request merged**